### PR TITLE
bpo-34596: Fallback to a default reason when @unittest.skip is uncalled

### PR DESCRIPTION
**Pull request by Naitreey**
-   Provide a default reason string when `@unittest.skip` decorator is used in uncalled form.
-   Add a test case for this change.

https://bugs.python.org/issue34596

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Review by zware — APPROVED** [review]
LGTM.  I would like to hear from the maintainers of unittest before merging, though.

File: Lib/unittest/case.py
Comment (berkerpeksag) [inline Lib/unittest/case.py]: I think it would be better if set `reason` to `None`.

File: Misc/NEWS.d/next/Tests/2018-09-07-01-18-27.bpo-34596.r2-EGd.rst
Comment (berkerpeksag) [inline Misc/NEWS.d/next/Tests/2018-09-07-01-18-27.bpo-34596.r2-EGd.rst]: @unittest.skip -> :func:\`unittest.skip\`

**Review by berkerpeksag — COMMENTED** [review]
https://docs.python.org/3/library/unittest.html#unittest.skip documentations needs to be updated as well.

File: Lib/unittest/case.py
Comment (kushaldas) [reply Lib/unittest/case.py]: I would prefer to have the reason to `""`, this will be also [similar](http://junit.sourceforge.net/javadoc/org/junit/Ignore.html) to the Java world.

**Review by kushaldas — CHANGES_REQUESTED** [review]

File: Lib/unittest/case.py
Comment (zware) [reply Lib/unittest/case.py]: I'm fine with that.  I suggested "unconditionally" mostly just because I had no better suggestion; the empty string probably should have been what I suggested in the first place :)

**Comment by Naitreey** [comment]
I have made the requested changes; please review again. Except for the documentation part. From the ongoing discussion on bug tracker, I'm not sure this change is strictly for not breaking existing user code, or intended as a new feature.

**Comment by serhiy-storchaka** [comment]
Even if this proposition be accepted, it is a new feature and should not be backported.

**Review by berkerpeksag — CHANGES_REQUESTED** [review]
Looking at this again, now I agree with Serhiy's suggestion at https://bugs.python.org/issue34596#msg324836

All skip*() APIs require 'reason' and if we'd accept this PR as-is, I'm sure we are going to get "why don't you make reason optional in skipIf() as well?" reports. I prefer to keep these APIs simpler.

**Comment by voidspace** [comment]
I'm going to land this PR as is on 3.8 and master.

**Pull request merged**
